pkg/agent/rest.go

```diff
@@ -34,27 +36,38 @@ func NewNodeZeroRestClient(ctx context.Context, assetDir string) (*NodeZeroRestC
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to create asset store")
 	}
-	nmState := &manifests.NMStateConfig{}
-	if err := assetStore.Fetch(nmState); err != nil {
-		return nil, errors.Wrapf(err, "failed to fetch %s", nmState.Name())
+
+	agentConfig, agentConfigError := assetStore.Load(&agentconfig.AgentConfig{})
+	agentManifests, manifestError := assetStore.Load(&manifests.AgentManifests{})
+
+	if agentConfigError != nil {
+		logrus.Debug(errors.Wrapf(agentConfigError, "failed to load %s", agentConfig.(*agentconfig.AgentConfig).Name()))
+	}
+	if manifestError != nil {
+		logrus.Debug(errors.Wrapf(manifestError, "failed to load %s", agentManifests.(*manifests.AgentManifests).Name()))
 	}
 
-	NodeZeroIP, err := manifests.GetNodeZeroIP(nmState.Config)
+	if agentConfigError != nil && manifestError != nil {
+		return nil, errors.New("failed to load AgentConfig and NMStateConfig")
+	}
+
+	RendezvousIP, err := image.RetrieveRendezvousIP(agentConfig.(*agentconfig.AgentConfig).Config, agentManifests.(*manifests.AgentManifests).NMStateConfigs)
```

Comment:
We should check that agentConfig and agentManifests [are non-nil](https://github.com/lranjbar/installer/blob/63c6e934ee6e33a4f2a5c63665a959aab58ca2a0/pkg/asset/store/store.go#L370) before dereferencing them here.